src/libsystemd/sd-network/network-util.c

```diff
@@ -56,3 +56,48 @@ static const char* const link_address_state_table[_LINK_ADDRESS_STATE_MAX] = {
 };
 
 DEFINE_STRING_TABLE_LOOKUP(link_address_state, LinkAddressState);
+
+int parse_operational_state_range(const char *str, LinkOperationalStateRange *out) {
+        LinkOperationalStateRange range;
+        _cleanup_free_ const char *min = NULL;
+        const char *p;
+
+        assert(str);
+        assert(out);
+
+        p = strchr(str, ':');
+        if (p) {
+                min = strndup(str, p - str);
+
+                if (isempty(p + 1))
+                        range.max = _LINK_OPERSTATE_INVALID;
+                else {
+                        range.max = link_operstate_from_string(p + 1);
+                        if (range.max < 0)
+                                return -EINVAL;
+                }
+        } else {
+                min = strdup(str);
+                range.max = _LINK_OPERSTATE_INVALID;
+        }
+
+        if (!min)
+                return -ENOMEM;
+
+        if (isempty(min))
+                range.min = _LINK_OPERSTATE_INVALID;
+        else {
+                range.min = link_operstate_from_string(min);
+                if (range.min < 0)
+                        return -EINVAL;
+        }
+
+        if (range.min != _LINK_OPERSTATE_INVALID &&
+                range.max != _LINK_OPERSTATE_INVALID &&
+                range.min > range.max)
```

Comment:
Indentatino.

Reply:
I'm assuming you meant continuation indents should also be 8 spaces. Please let me know if you meant something else (I looked at the style guide but didn't immediately see anything about continuation indents).

Reply:
We just generally follow what emacs does.  In this case, lines 96–97 should be indented to ling up with "range" above.